### ISSUE TITLE
Unity Ads doesn't show in new AdMob SDK (7.3.0+) 2023

### ISSUE THREAD

**Issue by Prudentibus** [googleads/googleads-mobile-unity#2644]
### [REQUIRED] Step 1: Describe your environment

  * Unity version: 2021.3.15f1
  * Google Mobile Ads Unity plugin version: v7.4.0 - 8.0.0
  * Platform: Android & Unity Editor
  * Platform OS version: Windows 10, Android API level 33
  * Any specific devices issue occurs on: Any
  * Mediation ad networks used, and their versions: None

### [REQUIRED] Step 2: Describe the problem

 I tried to use GoogleAdMob SDK for Unity from version 7.4.0 to 8.0.0, to show add, it Unity Editor it works perfectly fine.
 When game was build on any mobile devices ad doesn't show, but does ask for reqeust and fills with no errors.
 Ad might be show once in ~100 clicks, otherwise it doesn't do nothing, just request.
 I tried everything. Everything seems okay, but the problem, that it doesn't show AD on request, only filling it. 
 I need to mention, that I don't think the problem is that Ad Doesn't upload, before I ask to show it, since I tried version "Wthi Uploading Before".
 
 As I might see, this is common problem in internet.
 
 I fixed it by using version 7.1.0 with different implementation method.

#### Steps to reproduce:

Just followed this instruction:
https://developers.google.com/admob/unity/quick-start

And reproduce each step for rewarded/intertistial ad.

All works perfectly in Unity Editor.

But on build it doesn't works in either separate build for a phone or Google Play Market.

Just try to request and show the reward, and it does ask for request, but doesn't show ad.

#### Relevant Code:

```
private RewardedAd rewardedAd;

    private void Awake()
    {
       // I preload RewardedAd
        LoadRewardedAd();
    }

    /// <summary>
    /// Loads the rewarded ad.
    /// </summary>
    public void LoadRewardedAd()
    {
        // Clean up the old ad before loading a new one.
        if (rewardedAd != null)
        {
            rewardedAd.Destroy();
            rewardedAd = null;
        }

        Debug.Log("Loading the rewarded ad.");

        // create our request used to load the ad.
        var adRequest = new AdRequest.Builder().Build();

        // send the request to load the ad.
        RewardedAd.Load(_adUnitId, adRequest,
            (RewardedAd ad, LoadAdError error) =>
            {
                // if error is not null, the load request failed.
                if (error != null || ad == null)
                {
                    Debug.LogError("Rewarded ad failed to load an ad " +
                                   "with error : " + error);
                    return;
                }

                Debug.Log("Rewarded ad loaded with response : "
                          + ad.GetResponseInfo());

                rewardedAd = ad;
            });
    }

    public void ShowRewardedAd()
    {
        const string rewardMsg =
            "Rewarded ad rewarded the user. Type: {0}, amount: {1}.";

        if (rewardedAd != null && rewardedAd.CanShowAd())
        {
        rewardedAd.Show((Reward reward) =>
        {
                // TODO: Reward the user.
                Debug.Log(string.Format(rewardMsg, reward.Type, reward.Amount));
        });

        }
    }

    public void ActivateAd(string adID)
    {
      // I call this function from other script with specific UnitAdID.
        _adUnitId = adID;

       //Loading Rewarded Ad with new ID
        LoadRewardedAd();
        
        // Testing if the problem was that Ads isn't ready before showing it ( It wasn't the problem )
        //Invoke("ShowRewardedAd", 0.9f);
        
        // Show the rewarded AD - This is the main problem, by some reason it doesn't show, no matter what should I do.
        ShowRewardedAd();
    }
```

**Comment by Prudentibus** [comment]
It feels like I forgot something before "ShowAd", but it doesn't seems like it by looking to sample scripts.
Maybe I am wrong?

**Comment by NVentimiglia** [comment]
@Prudentibus 

1.  Make sure you set your publisher id and call 'MobileAds.Initialize()` before loading an ad as described in [get started](https://developers.google.com/admob/unity/quick-start#set_your_admob_app_id)
2. If you still have issues, please include some debug logging so I can identify the issue.

**Comment by Prudentibus** [comment]
> @Prudentibus
> 
>     1. Make sure you set your publisher id and call 'MobileAds.Initialize()` before loading an ad as described in [get started](https://developers.google.com/admob/unity/quick-start#set_your_admob_app_id)
> 
>     2. If you still have issues, please include some debug logging so I can identify the issue.

Yes I did both thing naturally.
And as I said, I made it work with 7.1.0 Version.

**Comment by mdobele** [comment]
To me it just looks like the problem is you're just not waiting for the ad to load and are trying to call ShowRewardedAd() immediately after calling LoadRewardedAd().  Even your test of delaying it with that  //Invoke("ShowRewardedAd", 0.9f); wouldn't be enough as it can take 10+ seconds on device sometimes.  It wold work on Unity Editor since it's not actually fetching real adverts and is showing fake ones. 

You should be relying on your OnAdLoaded callback to know when the ad has returned successfully and then only calling ShowRewarded() after that has happened.

**Comment by Prudentibus** [comment]
> To me it just looks like the problem is you're just not waiting for the ad to load and are trying to call ShowRewardedAd() immediately after calling LoadRewardedAd(). Even your test of delaying it with that //Invoke("ShowRewardedAd", 0.9f); wouldn't be enough as it can take 10+ seconds on device sometimes. It wold work on Unity Editor since it's not actually fetching real adverts and is showing fake ones.
> 
> You should be relying on your OnAdLoaded callback to know when the ad has returned successfully and then only calling ShowRewarded() after that has happened.

By this logic preload ad should work perfectly fine, right? ( On test device it shown, that ad was loaded )
But it doesn't.
I might assume the problem is that it's loading to fast, but I have no idea why it would much faster on 7.1.0 then.

**Comment by Jackyiyo** [comment]
Hi, I am facing the same issue. Are you solved that issue? or just fixed it with 7.1.0 Google admobs external addons?

**Comment by Prudentibus** [comment]
> Hi, I am facing the same issue. Are you solved that issue? or just fixed it with 7.1.0 Google admobs external add-ons?

Yeah, there were several issues:
 1. By some reason Ads upload time takes 6000ms+; It fixed after a while by itself
 2. I used to upload ads while watching ads, or before starting level, but it not always worked probably because of 1 problem.
 
 So, the solution I came with for 8.0.0 is:
  1. Create a list of ads I want to upload for each section, at least 3 of them, whenever User ask for Ads to show, he sees already preloaded ad, and new added added to list. SO there always several ads in a queue to show. ( Also you can make gameObject "Don'tDestroyOnLoad" - to be sure Ads doesn't take long time to upload. )
